### PR TITLE
maturin: replace name by pname

### PR DESCRIPTION
**Pull request by danieldk**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/maturin/default.nix [hunk 1]
```diff
@@ -4,7 +4,7 @@
 let
   inherit (darwin.apple_sdk.frameworks) Security;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/maturin/default.nix:5]: Normally Security belongs into inputs and is inherited from all packages.

Comment (danieldk) [reply pkgs/development/tools/rust/maturin/default.nix]: Fixed. Also reformatted the derivation with `nixpkgs-fmt` and removed the `gmp` and `openssl` buildInputs, which are not used anymore.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112775` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
40 packages built:
</summary>
<ul>
<li>maturin</li>
<li>python37Packages.fastapi</li>
<li>python37Packages.fastdiff</li>
<li>python37Packages.graphene</li>
<li>python37Packages.johnnycanencrypt</li>
<li>python37Packages.qiskit</li>
<li>python37Packages.qiskit-aer</li>
<li>python37Packages.qiskit-aqua</li>
<li>python37Packages.qiskit-ibmq-provider</li>
<li>python37Packages.qiskit-ignis</li>
<li>python37Packages.qiskit-terra</li>
<li>python37Packages.retworkx</li>
<li>python37Packages.snapshottest</li>
<li>python37Packages.starlette</li>
<li>python37Packages.tumpa</li>
<li>python37Packages.wasmer</li>
<li>python38Packages.fastapi</li>
<li>python38Packages.fastdiff</li>
<li>python38Packages.graphene</li>
<li>python38Packages.johnnycanencrypt</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aer</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-ignis</li>
<li>python38Packages.qiskit-terra</li>
<li>python38Packages.retworkx</li>
<li>python38Packages.snapshottest</li>
<li>python38Packages.starlette</li>
<li>python38Packages.tumpa</li>
<li>python38Packages.wasmer</li>
<li>python39Packages.fastdiff</li>
<li>python39Packages.graphene</li>
<li>python39Packages.johnnycanencrypt</li>
<li>python39Packages.retworkx</li>
<li>python39Packages.snapshottest</li>
<li>python39Packages.starlette</li>
<li>python39Packages.tumpa</li>
<li>python39Packages.wasmer</li>
<li>qutebrowser</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112775` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
18 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.fastapi</li>
<li>python37Packages.fastdiff</li>
<li>python37Packages.graphene</li>
<li>python37Packages.snapshottest</li>
<li>python37Packages.starlette</li>
<li>python37Packages.tumpa</li>
<li>python38Packages.fastapi</li>
<li>python38Packages.fastdiff</li>
<li>python38Packages.graphene</li>
<li>python38Packages.snapshottest</li>
<li>python38Packages.starlette</li>
<li>python38Packages.tumpa</li>
<li>python39Packages.fastapi</li>
<li>python39Packages.fastdiff</li>
<li>python39Packages.graphene</li>
<li>python39Packages.snapshottest</li>
<li>python39Packages.starlette</li>
<li>python39Packages.tumpa</li>
</ul>
</details>
<details>

  <summary>8 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.qiskit: log was empty</li>
<li>python37Packages.qiskit-aer: log was empty</li>
<li>python37Packages.qiskit-aqua: log was empty</li>
<li>python37Packages.qiskit-ignis: log was empty</li>
<li>python38Packages.qiskit: log was empty</li>
<li>python38Packages.qiskit-aer: log was empty</li>
<li>python38Packages.qiskit-aqua: log was empty</li>
<li>python38Packages.qiskit-ignis: log was empty</li>
  </ul>
</details>
<details>
<summary>
12 packages built:
</summary>
<ul>
<li>maturin</li>
<li>python37Packages.johnnycanencrypt</li>
<li>python37Packages.qiskit-ibmq-provider</li>
<li>python37Packages.qiskit-terra</li>
<li>python37Packages.retworkx</li>
<li>python38Packages.johnnycanencrypt</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-terra</li>
<li>python38Packages.retworkx</li>
<li>python39Packages.johnnycanencrypt</li>
<li>python39Packages.retworkx</li>
<li>qutebrowser</li>
</ul>
</details>

**Pull request merged**